File: test_resources_output/application/IntroduceParameterObject/IntroduceParameterToComplicatedParametersTest.java
package application;

public class IntroduceParameterToComplicatedParametersTest {
	void m0(){
		
	}
	
	void m1(int a){
		System.out.println(a);
	}
	
	void m2(int a, int b){
		System.out.println(a + b);
	}
	
	void m3(M3Parameter parameterObject){
		System.out.println(parameterObject.a + parameterObject.b + parameterObject.c);
	}
	void m4(M4Parameter parameterObject){
		System.out.println(parameterObject.a + parameterObject.b + parameterObject.c + parameterObject.d);
	}
	void m5(M5Parameter parameterObject){
		System.out.println(parameterObject.a + parameterObject.b + parameterObject.c + parameterObject.d + parameterObject.e);
	}

}

File: refactoring_test/rscore/tests/application/IntroduceParameterToComplicatedParametersTest.scala
package rscore.tests.application
import org.junit.Test
import rscore.util.FileUtil
import rscore.dsl.entity.collection.By

class IntroduceParameterToComplicatedParametersTest extends ApplicationTest {
	// TODO: extract below helper methods
	def doAssert2(testName: String, ignoreComment: Boolean = true): Unit = {
		val actualSource = FileUtil.eliminateBlankLines(
			this.fgPackageP.getCompilationUnit(testName + ".java").getSource())

		val outputFilepath = "test_resources_output/" + testGroupIdentifier + "/" + "IntroduceParameterObject" + "/" + testName + ".java"
		val expectedSource = FileUtil.getFileContents(outputFilepath)

		doAssertHelper(expectedSource, actualSource, ignoreComment)
	}

	@Test
	def 特定数以上のパラメータを持つ関数にパラメータオブジェクトを導入する(): Unit = {
		val testName = "IntroduceParameterToComplicatedParametersTest"
		prepareTest(testName)

		val threshold = 3
		val methods = getFirstClass().methods().filter(_.parameters().count >= threshold)

		methods.foreach(_.introduceParameterObject())

		doAssert2(testName) // test for original

		// tests for generated parameterobject classes
		(3 to 5).foreach(i => {
			val pObjName = "M%sParameter".format(i)
			val actualSource = FileUtil.eliminateBlankLines(getCurrentProject().pkg("application").classes.selectOne(By name pObjName).origin().getCompilationUnit().getSource())
			val expectedSource = FileUtil.getFileContents("test_resources_output/application/IntroduceParameterObject/" + pObjName + ".java", true)
			doAssertHelper(expectedSource, actualSource, true)
		})
	}

}